src/tvheadend/ChannelTuningPredictor.cpp

```diff
@@ -74,7 +74,7 @@ uint32_t ChannelTuningPredictor::PredictNextChannelId(uint32_t tuningFrom, uint3
    * it will point at the channel we should tune to */
   std::set<ChannelPair>::iterator predictedIt = m_channels.cend();
 
-  if (std::next(fromIt, 1) == toIt || toIt->second == firstNum) {
+  if (fromIt == m_channels.cend() || std::next(fromIt, 1) == toIt || toIt->second == firstNum) {
```

Comment:
If `fromIt` is equal to the end iterator it would mean that `GetIterator` doesn't find the channel in question which would mean the root cause lies somewhere else. `fromIt` and `toIt` being valid iterators are supposed to be invariants.